lib/lumberjack/client.rb

```diff
@@ -114,18 +109,28 @@ def write(msg)
     end
 
     public
-    def write_hash(hash)
-      frame = Encoder.to_compressed_frame(hash, inc)
-      ack if unacked_sequence_size >= @window_size
-      write frame
+    def write_hash(elements)
+      elements = [elements] if elements.is_a?(Hash)
+      send_window_size(elements.size)
+
+      payload = elements.map { |element| Encoder.to_frame(element, inc) }.join
+      compress = compress_payload(payload)
+      send_payload(compress)
+
+      ack(elements.size)
+    end
+
+    private 
+    def compress_payload(payload)
+      compress = Zlib::Deflate.deflate(payload)
+      ["1", "C", compress.bytesize, compress].pack("AANA#{compress.length}")
```

Comment:
I haven't revised the actual code, I'll change it for `A*`.
I guess this would not have been an issue because the `.length` was done on a compressed string and would make it UTF-8 free? Right?